### PR TITLE
Feature/typed query

### PR DESCRIPTION
**Pull request by ohadravid**
Adds support for queries which deserialize results according to the requested return value,

Raw query, with user defined text.
```
let results: Vec<HashMap<String, Variant>> = wmi_con.raw_query("SELECT Name, Debug FROM Win32_OperatingSystem");
```

Query by struct.
```
let results: Vec<Win32_OperatingSystem> = wmi_con.query();
```

Filtered query (minimal implementation).
```
let mut filters = HashMap::new();
filters.insert("Name".to_owned(), FilterValue::Str("cargo.exe"));
let results = wmi_con.filtered_query::<Win32_Process>(&filters);
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -10,6 +10,9 @@ pub mod datetime;
 pub mod safearray;
 pub mod consts;
 
+//#[cfg(test)]
```
Comment (omerbenamram) [inline src/lib.rs]: put it back?

File: src/query.rs [hunk 2]
```diff
@@ -1,29 +1,94 @@
...
 };
 
+pub enum FilterValue {
+    Bool(bool),
+    Number(i64),
+    Str(&'static str),
+    String(String),
+}
+
+fn build_query<'de, T>(filters: Option<&HashMap<String, FilterValue>>) -> String
```
Comment (omerbenamram) [inline src/query.rs]: docs 🙏

File: src/query.rs [hunk 3]
```diff
@@ -191,4 +314,76 @@ mod tests {
...
+    #[test]
+    fn it_builds_correct_query() {
+        #[derive(Deserialize, Debug)]
+        struct Win32_OperatingSystem {
+            Caption: String,
+        }
+
+        let mut filters = HashMap::new();
+
+        filters.insert("C1".to_string(), FilterValue::Str("a"));
```
Comment (omerbenamram) [inline src/query.rs]: why can't "C1" accept 'static str?

File: src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+use crate::WMIConnection;
+
+pub mod fixtures {
+    use super::*;
+    use lazy_static::lazy_static;
+    use std::rc::Rc;
+
+    // This way we only setup COM security once during tests.
+    lazy_static! {
+        static ref COM_LIB: COMLibrary = { COMLibrary::new().unwrap() };
```
Comment (omerbenamram) [inline src/tests.rs]: you could probably get away with `https://doc.rust-lang.org/std/sync/struct.Once.html`

File: src/query.rs [hunk 5]
```diff
@@ -315,6 +315,30 @@ mod tests {
...
+    #[test]
+    fn it_fails_gracefully() {
+        let wmi_con = wmi_con();
+
+        let enumerator = wmi_con
+            .exec_query_native_wrapper("SELECT NoSuchField FROM Win32_OperatingSystem")
+            .unwrap();
+
+        for res in enumerator {
+            assert!(res.is_err())
```
Comment (omerbenamram) [inline src/query.rs]: show how to access the concrete underlying errors?

File: src/query.rs (hunk 3 above)
Comment (ohadravid) [reply src/query.rs]: It can (in theory), but it's more code so I just defer it to when we would actually need it

**Pull request merged**